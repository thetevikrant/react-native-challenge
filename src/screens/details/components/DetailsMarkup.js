import * as React from "react";
import { Image } from "react-native";
import {
  Container,
  Header,
  Title,
  Text,
  Button,
  Icon,
  Left,
  Body,
  Right,
  Thumbnail,
  Card,
  CardItem,
} from "native-base";

import styles from "./styles";
export interface Props {
  navigation: any;
  data: any;
}
export interface State {}
class DetailsMarkup extends React.Component<Props, State> {
  render() {
    const { data, navigation } = this.props;
    return (
      <Container>
        <Header>
          <Left>
            <Button
              transparent
              onPress={() => navigation.navigate("Home")}
            >
              <Icon name="arrow-back" />
            </Button>
          </Left>
          <Body>
            <Title>Details</Title>
          </Body>
          <Right />
        </Header>
        <Card style={styles.detailsCard}>
          <CardItem>
            <Left>
              <Thumbnail source={{uri: "https://cdn3.iconfinder.com/data/icons/business-bundle/512/news-512.png"}} />
              <Body>
                <Text>{data.title}</Text>
                <Text note>April 15, 2016</Text>
              </Body>
            </Left>
          </CardItem>
          <CardItem>
            <Body>
              <Image style={styles.detailsImage} source={{uri: data.img}} />
              <Text style={styles.detailsImageText}>{data.description}</Text>
            </Body>
          </CardItem>
        </Card>
      </Container>
    );
  }
}

export default DetailsMarkup;
