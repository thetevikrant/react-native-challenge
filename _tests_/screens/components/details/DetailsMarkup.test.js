import React from "react";
import DetailsMarkup from "../../../../src/screens/details/components/DetailsMarkup";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

const navigation = { navigate: jest.fn() };
const data = { map: jest.fn() };

it("renders correctly", () => {
  const tree = renderer.create(<DetailsMarkup navigation={navigation} data={data} />).toJSON();
  expect(tree).toMatchSnapshot();
});