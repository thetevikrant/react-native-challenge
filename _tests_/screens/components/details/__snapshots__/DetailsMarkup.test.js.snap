// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderBottomColor": "#a7a6ab",
          "borderBottomWidth": 0.5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 15,
          "right": 0,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
          "top": 0,
        }
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          }
        }
        virtual={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": null,
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginLeft": -3,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 6,
              "paddingRight": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#007aff",
                    "fontSize": 33,
                    "marginLeft": 2,
                    "marginRight": 2,
                    "marginTop": 2,
                    "paddingTop": 2,
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            virtual={undefined}
          >
            Ôèè
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "600",
              "textAlign": "center",
            }
          }
          virtual={undefined}
        >
          Details
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
        virtual={undefined}
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
          "borderLeftWidth": 0.5,
          "borderRadius": 2,
          "borderRightWidth": 0.5,
          "borderTopWidth": 0.5,
          "elevation": 3,
          "flex": 1,
          "flexWrap": "wrap",
          "marginBottom": 5,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 1.5,
        },
        Object {
          "flex": 0,
          "marginLeft": 5,
          "marginRight": 5,
        },
      ]
    }
    virtual={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
          }
        }
        virtual={undefined}
      >
        <Image
          source={
            Object {
              "uri": "https://cdn3.iconfinder.com/data/icons/business-bundle/512/news-512.png",
            }
          }
          style={
            Object {
              "borderRadius": 28,
              "height": 56,
              "width": 56,
            }
          }
          virtual={undefined}
        />
        <View
          style={
            Object {
              "alignItems": null,
              "alignSelf": "center",
              "flex": 1,
              "marginLeft": 10,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              }
            }
            uppercase={false}
            virtual={undefined}
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            note={true}
            style={
              Object {
                "color": "#808080",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "400",
                "marginRight": 20,
              }
            }
            uppercase={false}
            virtual={undefined}
          >
            April 15, 2016
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "stretch",
            "flex": 1,
          }
        }
        virtual={undefined}
      >
        <Image
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "height": 200,
              "width": 335,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "marginTop": 10,
              },
            ]
          }
          uppercase={false}
          virtual={undefined}
        />
      </View>
    </View>
  </View>
</View>
`;
